From 3ee3ce9c6f5bf9b979a8bc3505ade0b2adbdbeac Mon Sep 17 00:00:00 2001
From: Kevin Hao <kexin.hao@windriver.com>
Date: Wed, 1 Sep 2010 02:53:48 -0700
Subject: [PATCH] bman: init buffer pool even though not all cores are online

By default all the cores will be online, and each core has affine
portal. But sometimes we just want some of the cores to be online
by using command options "maxcpus=x". So we should also initialize
the buffer pool in this situation.

Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/hwalloc/bman_driver.c |   13 ++++++-------
 1 files changed, 6 insertions(+), 7 deletions(-)

diff --git a/drivers/hwalloc/bman_driver.c b/drivers/hwalloc/bman_driver.c
index 389b8a1..a592f1b 100644
--- a/drivers/hwalloc/bman_driver.c
+++ b/drivers/hwalloc/bman_driver.c
@@ -336,13 +336,12 @@ static __init int bman_init(void)
 			return ret;
 	}
 #ifndef CONFIG_FSL_BMAN_PORTAL_DISABLEAUTO
-	if (num_affine_portals == num_online_cpus()) {
-		for_each_compatible_node(dn, NULL, "fsl,bpool") {
-			int ret = fsl_bpool_init(dn);
-			if (ret)
-				return ret;
-		}
-	} else {
+	for_each_compatible_node(dn, NULL, "fsl,bpool") {
+		int ret = fsl_bpool_init(dn);
+		if (ret)
+			return ret;
+	}
+	if (num_affine_portals != num_online_cpus()) {
 		pr_err("Not all cpus have an affine Bman portal\n");
 		pr_err("Ignoring buffer pools\n");
 		pr_err("Expect Bman-dependent drivers to crash!\n");
-- 
1.6.5.2

